docs/nx-cloud/private/kubernetes-setup.md

```diff
@@ -0,0 +1,287 @@
+
+The result should look like
+this: `mongodb+srv://admin-user:DB_PASSWORD@cloud-mongodb-svc.default.svc.cluster.local/nrwl-api?replicaSet=cloud-mongodb&ssl=false`
+.
+
+Extract the connection string and paste it into your `secret.yml`.
+
+### Step 3: Create a secret
+
+Create a secret by running `kubectl apply -f examples/secret.yml`
+
+### Step 4: Install Nx Cloud using helm
+
+```
+> helm repo add nx-cloud https://nrwl.github.io/nx-cloud-helm
+> helm install nx-cloud nx-cloud/nx-cloud --values=overrides.yml
+```
+
+`examples/overrides` contains the min overrides files. You need to provision:
+
+1. The image tag you want to install
+2. `nxCloudAppURL` which is the url used to access ingress from CI and dev machines (
+   e.g., `https://nx-cloud.myorg.com`).
+3. `secret/name` the name of the secret you created in Step 3.
+4. `secret/nxCloudMongoServerEndpoint`, the name of the key from the secret.
+   5`secret/adminPassword`, the name of the key from the secret.
+
+If you only applied the secret from Step 3, the only thing you will need to change is `nxCloudAppURL`.
+
+## Cloud Containers
+
+The installation will create the following:
+
+1. nx-cloud-frontend (deployment)
+2. nx-cloud-api (deployment)
+3. nx-cloud-nx-api (deployment)
+4. nx-cloud-file-server (deployment)
+5. nx-cloud-aggregator (cron job)
+
+## Ingress, IP, Certificates
+
+You can configure Ingress. For instance, the following will see the ingress class to 'gce', the global static ip name
+to 'nx-cloud-ip', and will set a global Google managed certificate.
+
+```yaml
+image:
+  tag: 'latest'
+
+nxCloudAppURL: 'https://nx-cloud.myorg.com'
+
+ingress:
+  class: 'gce'
+  globalStaticIpName: 'nx-cloud-ip'
+  managedCertificates: 'cloud-cert'
+
+secret:
+  name: 'cloud'
+  nxCloudMongoServerEndpoint: 'NX_CLOUD_MONGO_SERVER_ENDPOINT'
+  adminPassword: 'ADMIN_PASSWORD'
+```
+
+This configuration will look different for you. You will have a different global static ip and your cert name will also
+be different. If you are interested in creating the two using GKE, check out the following links:
+
+- [Reserving a static external IP address](https://cloud.google.com/compute/docs/ip-addresses/reserve-static-external-ip-address)
+- [Using Google-managed SSL certificates](https://cloud.google.com/kubernetes-engine/docs/how-to/managed-certs)
+
+If you aren't using GKE, `ingress.class` will also be different. For example, see [our example config for AWS](https://github.com/nrwl/nx-cloud-helm/blob/main/aws-guide/helm-values.yml#L7) or check out the AWS Load Balancer set-up section [here for AWS set-up instructions.](https://github.com/nrwl/nx-cloud-helm/aws-guide/AWS-GUIDE.md#3-install-a-load-balancer)
+
+If you need to have a detailed Ingress configuration, you can tell the package to skip defining ingress:
+
+```yaml
+image:
+  tag: 'latest'
+
+nxCloudAppURL: 'https://nx-cloud.myorg.com'
+
+ingress:
+  skip: true
+```
+
+<details>
+<summary>⤵️ and then define it yourself (expand me)</summary>
```

Comment:
The <details> and <summary> tags aren't properly rendered on nx.dev. We can probably just remove them and show the full content.